onnxruntime/test/fuzzing/src/test.cpp

```diff
@@ -265,143 +239,170 @@ int processCommandLine(int argc, char* argv[], runtimeOpt& opt)
+      //
+      if (model_proto.ParseFromIstream(&modelStream)) {
+        if (repo_mode) {
+          Logger::testLog << L"Running Prediction for: " << model_file_name
+                          << L" with seed " << seed << Logger::endl;
+          mutateModelTest(model_proto, mutate_model_dir_name, user_opt, env, seed);
+          Logger::testLog << L"Finished Prediction for: " << model_file_name
+                          << L" with seed " << seed << Logger::endl;
+          return 0;
+        } else {
+          // Call the mutateModelTest
           //
-          currTime = std::chrono::system_clock::now();
-        }
-        Logger::testLog<< "Ending Test" << Logger::endl;
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.enable();
+          std::chrono::system_clock::time_point curr_time{std::chrono::system_clock::now()};
+
+          std::chrono::minutes time_in_min{test_time_out};
+          std::chrono::seconds time_in_sec{test_time_out};
+          std::chrono::hours time_in_hrs{test_time_out};
+          std::chrono::system_clock::time_point end_time{curr_time};
+          end_time += scale == timeScale::Hrs ? time_in_hrs
+                                              : scale == timeScale::Min ? time_in_min : time_in_sec;
+          Logger::testLog << "Starting Test" << Logger::endl;
+          while (curr_time < end_time) {
+            try {
+              onnx::ModelProto bad_model = model_proto;
+              Logger::testLog << "Starting Test iteration: " << run_stats.iteration << Logger::endl;
+              mutateModelTest(bad_model, mutate_model_dir_name, user_opt, env);
+              run_stats.num_successful_runs++;
+              Logger::testLog << "Completed Test iteration: " << run_stats.iteration++ << Logger::endl;
+            } catch (...) {
+              fuzz_handle_exception(run_stats);
+            }
+            // Update current time
+            //
+            curr_time = std::chrono::system_clock::now();
+          }
         }
-
-        Logger::testLog<< L"Total number of exceptions: " << num_unknown_exception+num_std_exception+num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of Unknown exceptions: " << num_unknown_exception << Logger::endl;
-        Logger::testLog<< L"Number of ort exceptions: " << num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of std exceptions: " << num_std_exception << Logger::endl;
-        Logger::testLog << L"Number of unique errors: "<< werrStreamBuf.get_unique_errors() << L"\n";
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.disable();
-          Logger::testLog.flush();
+      } else {
+        throw std::exception("Unable to initialize the Onnx model in memory");
+      }
+    } else {
+      std::wstring ort_model_file = model_file;
+      if (model_file.substr(model_file.find_last_of(L".") + 1) == L"onnx") {
+        ort_model_file = model_file + L".ort";
```

Comment:
Can we make the extension checks case insensitive?

Also add comment saying that the code here is generating an ORT format model from the ONNX model as that won't be obvious unless you know a lot about ORT.